lib/realtime/api.ex

```diff
@@ -225,4 +238,25 @@ defmodule Realtime.Api do
     |> Map.put(:events_per_second_rolling, avg)
     |> Map.put(:events_per_second_now, current)
   end
+
+  defp maybe_invalidate_cache(%Ecto.Changeset{changes: changes, valid?: true, data: %{external_id: external_id}})
+       when changes != %{} do
+    Tenants.Cache.invalidate_tenant_cache(external_id)
```

Comment:
Should this be calling `distributed_invalidate_tenant_cache` ? Otherwise a remote node won't have the new settings when `Connect.lookup_or_start_connection` is called the next time. 

And that's what I mean by race condition:

* Tell all nodes to invalidate cache (broadcast)
* Tell Connect to shutdown (broadcast)

There is no real guarantee which one of the two above will be processed first on each node ^

Then at some point `Connect.lookup_or_start_connection` will be called and it's not 100% clear that the new settings will be available if this makes sense.

🤔

Reply:
honestly I have no clue what to do because there's no good way of guaranteeing anything 😅

Reply:
Sounds good!